### PR TITLE
*: remove localstore dependency.

### PR DESCRIPTION
**Pull request by hanfei1991**

### CODE REVIEW

**Comment by coocood** [comment]
leveldb is used in tidb-test

File: meta/autoid/autoid.go [hunk 1]
```diff
@@ -56,6 +60,21 @@ func GetStep() int64 {
...
+	step = s
+}
+
+// Base implements autoid.Allocator Base interface.
+func (alloc *allocator) Base() int64 {
+	return alloc.base
+}
+
+// End implements autoid.Allocator End interface.
+func (alloc *allocator) End() int64 {
```
Comment (coocood) [inline meta/autoid/autoid.go]: Why do we need `End` if it the same as `Base`?

Comment (hanfei1991) [reply meta/autoid/autoid.go]: My fault, it should return end

**Comment by hanfei1991** [comment]
/run-all-test=pr/388

**Comment by hanfei1991** [comment]
@coocood PTAL

File: bootstrap_test.go [hunk 2]
```diff
@@ -44,7 +44,6 @@ func (s *testBootstrapSuite) TestBootstrap(c *C) {
 	defer testleak.AfterTest(c)()
 	store, dom := newStoreWithBootstrap(c, s.dbName)
 	defer dom.Close()
-	defer store.Close()
```
Comment (zz-jason) [inline bootstrap_test.go]: why remove this ?

File: perfschema/init.go [hunk 3]
```diff
@@ -349,11 +349,19 @@ func (ps *perfSchema) initialize() {
 	ps.buildTables()
 }
 
-func (ps *perfSchema) GetDBMeta() *model.DBInfo {
+// GetDBMeta returns the db info.
```
Comment (zimulala) [inline perfschema/init.go]: s/db/DB

File: server/tidb_test.go [hunk 4]
```diff
@@ -43,7 +44,9 @@ var suite = new(TidbTestSuite)
 var _ = Suite(suite)
 
 func (ts *TidbTestSuite) SetUpSuite(c *C) {
-	store, err := tidb.NewStore("memory:///tmp/tidb")
+	store, err := tikv.NewMockTikvStore()
+	tidb.SetStatsLease(0)
+	tidb.SetSchemaLease(0)
```
Comment (zimulala) [inline server/tidb_test.go]: I think this code can be removed. Because we use mock `syncer`.

Comment (hanfei1991) [reply server/tidb_test.go]: Great!

File: bootstrap_test.go (hunk 2 above)
Comment (hanfei1991) [reply bootstrap_test.go]: because store has been closed in line 97

**Comment by hanfei1991** [comment]
@zimulala @zz-jason PTAL

**Comment by coocood** [comment]
LGTM

**Pull request merged**